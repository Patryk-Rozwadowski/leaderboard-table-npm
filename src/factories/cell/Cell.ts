import ElementCreator, { ComponentFactory } from "../creators/ElementCreator";
import { SingleCellProperties } from "../../common/common.types";
import DOMController from "../../controllers/DOMController";
import { SEMANTIC_TAGS } from "../../style/semanticTags";
import { COMPONENT_STYLES } from "../../style/styleClasses";

class Cell implements ComponentFactory<Cell> {
   DOMController: DOMController;

   private _elementCreator: ElementCreator;
   private _rowListContainer: HTMLElement;

   constructor(
      private _rootContainer: HTMLElement,
      private _rowData: SingleCellProperties
   ) {
      this._elementCreator = new ElementCreator();
      this.DOMController = new DOMController();
   }

   public create(): HTMLElement {
      this._rowListContainer = this._createCellContainer();
      this._createCell();
      return this._rowListContainer;
   }

   private _createCellContainer() {
      return this._elementCreator.createContainer(
         SEMANTIC_TAGS.CELL_CONTAINER,
         COMPONENT_STYLES.CELL_CONTAINER
      );
   }

   private _createCell() {
      const contentContainer = this._createRowContentContainer();
      const textContent = this._rowContentText();
      DOMController.appendChildrenToContainer(contentContainer, textContent);
      this._rowListContainer.appendChild(contentContainer);
   }

   private _rowContentText(): HTMLElement {
      return this._elementCreator.createText(
         SEMANTIC_TAGS.PRIMARY_TEXT,
         this._rowData.toString()
      );
   }

   private _createRowContentContainer(): HTMLElement {
      return this._elementCreator.createContainer(
         SEMANTIC_TAGS.CELL_CONTAINER,
         COMPONENT_STYLES.CELL
      );
   }
}

export default Cell;
