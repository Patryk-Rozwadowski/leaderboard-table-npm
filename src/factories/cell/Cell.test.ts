import Cell from "./Cell";

describe("Cell component", () => {
   const cellData = "name";
   let root: HTMLElement;
   let cell: Cell;
   beforeEach(() => {
      root = document.createElement("div");
      root.classList.add(".leaderboard-root");
      cell = new Cell(root, cellData);
   });

   it("should be created and appended to it's container", () => {
      expect(cell.create()).toMatchSnapshot();
   });

   it("container should be created", () => {
      expect(cell["_createCellContainer"]()).toMatchSnapshot();
   });

   it("text content element should be created", () => {
      expect(cell["_cellContentText"]()).toMatchSnapshot();
   });

   it("content container should be created", () => {
      expect(cell["_createCellContainer"]()).toMatchSnapshot();
   });
});
