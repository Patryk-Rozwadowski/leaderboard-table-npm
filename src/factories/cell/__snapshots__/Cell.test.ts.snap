// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cell component container should be created 1`] = `
<div
  class="lb_cell_container"
/>
`;

exports[`Cell component content container should be created 1`] = `
<div
  class="lb_cell_container"
/>
`;

exports[`Cell component should be created and appended to it's container 1`] = `
<div
  class="lb_cell_container"
>
  <div
    class="lb_cell"
  >
    <p
      class="lb_text_primary"
    >
      name
    </p>
  </div>
</div>
`;

exports[`Cell component text content element should be created 1`] = `
<p
  class="lb_text_primary"
>
  name
</p>
`;
